### PR TITLE
Allow overriding the wkid in calls to Esri services

### PR DESCRIPTION
**Pull request by gwendolyngoetz**
### Description

The Esri API has the ability to return spatial reference data using different coordinate systems via a wellKnownId (a.k.a wkid). Details of that along with links to the long list of possible coordinate systems can be found on the [arcgis developer site](https://developers.arcgis.com/rest/services-reference/enterprise/using-spatial-references.htm).

The default wellKnownId is `4326 - GPS` which are traditional latitude and longitude value most often thought around of around coordinate systems. Custom hosted installations may set a different value as their default. For example, Seattle's installation defaults to using wellKnownId `2926 - NAD_1983_HARN_StatePlane_Washington_North_FIPS_4601_Feet` [link](https://developers.arcgis.com/rest/services-reference/enterprise/pdf/pcs_PDF_11.2.pdf).

This works fine as long as calls to correct an address and calls to mapping layers are defaulting to the same wellKnownId. Such as calling both endpoints on Seattle's instance.

Where it becomes a problem is if you want to start mixing calls to different installations.

#### Example

> 
> - Call arcgis.com to correct address, returns
> 
>   ```json5
>   // A very much truncated example
>   {
>    "spatialReference": {
>     "wkid": 4326
>    },
>    "candidates": [{
>      "location": {
>       "x": -122.329922012207,
>       "y": 47.605043979178
>      },
>     }]
>   }
>   ```
> 
> - Call Seattle map query endpoint passing in `x`, `y`, and `wkid` values from previous arcgis.com response. By no match is found because the coordinates are in 4326 format and the service expects 2926 format.
> 
> - Call King County map query endpoint passing in `x`, `y`, and `wkid` values from previous arcgis.com response. By no match is found because the coordinates are in 4326 format and the service expects 2926 format.

We can normalize this by specifying the output spatial reference when we call to correct the address using the `outSR` parameter.

#### Existing data

After we get the corrected address we save the x, y, and wkid as part of the address. Calls to map query endpoints already use the stored wkid to specify what the input spatial reference should be. Existing records will still work as is. 

> [!WARNING]
> If the override is added at a later date and is different from the previously used default you will end up having new records in a different format. If that is a concern the solution is to make sure the override value is the same as used before.
>
> This is only an issue if you change the find address endpoint at a later point in time to one with a different default. Such as using Seattle's version to using arcgis.com
> 
> If there is only a single host this will never be a problem since they'll both have the same defaults.

## Release notes

Allow overriding the default Esri wellKnownId value to allow consistent spatial reference settings if using service area validation from different sources.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary

### Issue(s) this relates to

Relates to #6800

### CODE REVIEW

File: server/app/services/settings/SettingsManifest.java [hunk 1]
```diff
@@ -578,6 +578,18 @@ public Optional<Integer> getEsriExternalCallTries() {
     return getInt("ESRI_EXTERNAL_CALL_TRIES");
   }
 
+  /**
+   * Forces calls to Esri services to use the specified spatial reference wellKnownId value for the
+   * [coordinate
+   * system](https://developers.arcgis.com/rest/services-reference/enterprise/using-spatial-references.htm).
+   * If not set the default configuration from the Esri server is used. Setting this may be needed
```
Comment (dkatzz) [inline server/app/services/settings/SettingsManifest.java:585]: **question:** Is it worth noting the default as 4326 - GPS?

**Review by dkatzz — APPROVED** [review]
Overall the changes LGTM! Could we update the docs as well https://docs.civiform.us/it-manual/sre-playbook/initial-deployment/configure-gis-service?

**Comment by gwendolyngoetz** [comment]
> Overall the changes LGTM! Could we update the docs as well https://docs.civiform.us/it-manual/sre-playbook/initial-deployment/configure-gis-service?

@dkatzz  Already in progress

File: server/app/services/settings/SettingsManifest.java (hunk 1 above)
Comment (dkatzz) [reply server/app/services/settings/SettingsManifest.java:585]: Oh got it makes sense!

**Review by melanie-exygy — APPROVED** [review]
question (non-blocking):  Would we ever have a situation where two different mapping services (Seattle vs. King County, for example) expect a different wellKnownID?

**Comment by gwendolyngoetz** [comment]
> question (non-blocking): Would we ever have a situation where two different mapping services (Seattle vs. King County, for example) expect a different wellKnownID?

@melanie-exygy It's possible multiple ones could have different defaults, but we already are set up to tell the mapping layers (service area validation) what the source spatial reference is based on the value returned from correcting the address.

**Pull request merged**